app/src/main/java/ch/epfl/favo/common/CollectionWrapper.java

```diff
@@ -4,14 +4,29 @@
 
 public class CollectionWrapper {
 
-  private static final String TAG = "DatabaseWrapper";
   private String collectionReference;
 
   public CollectionWrapper(String collectionReference) {
     this.collectionReference = collectionReference;
   }
 
-  public void addDocument(String key, Map document) {
+  public void addDocument(String key, Map document) throws RuntimeException {
     DatabaseWrapper.addDocument(key, document, collectionReference);
   }
+
+  public void removeDocument(String key) throws RuntimeException {
+    DatabaseWrapper.removeDocument(key, collectionReference);
+  }
+
+  public void updateDocument(String key, Map<String, Object> updates) throws RuntimeException {
+    DatabaseWrapper.updateDocument(key, updates, collectionReference);
+  }
+
+  public void getDocument(String key, DocumentCallback callback) throws RuntimeException {
+    DatabaseWrapper.getDocument(key, collectionReference, callback);
+  }
+
+    public void getAllDocuments(MultipleDocumentsCallback callback) throws RuntimeException {
+        DatabaseWrapper.getAllDocuments(collectionReference, callback);
+    }
```

Comment:
Do we need a second wrapper on the DatabaseWrapper?

Reply:
Technically no, but I plan to refactor `CollectionWrapper` to use `User` and `Favor` classes instead of Maps of strings to objects.

Reply:
Good idea, so from my understanding CollectionWrapper will adapt the Map collections from the database to your actual data models. There is a weird indent at lines 29-31.